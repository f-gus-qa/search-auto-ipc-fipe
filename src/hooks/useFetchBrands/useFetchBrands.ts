import useAppActions from '../useAppActions';
import useFipeClient from '../useFipeClient';

function useFetchBrands() {
  const fipeClient = useFipeClient();
  const {
    setAlert,
    setBrands,
    setSelected,
    stopLoading,
    startLoading,
  } = useAppActions();

  const fetchBrands = async () => {
    setBrands([]);
    setSelected('brand', null);
    startLoading('field.brand');
    try {
      setBrands(await fipeClient.getBrands());
    } catch (e) {
      setAlert({
        type: 'error',
        message: 'Houve um erro ao pegar as marcas, tente novamente mais tarde',
      });
    } finally {
      stopLoading('field.brand');
    }
  };

  return fetchBrands;
}

export default useFetchBrands;
