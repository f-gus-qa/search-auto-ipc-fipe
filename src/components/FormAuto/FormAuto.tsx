import { useEffect } from 'react';
import {
  Box, Grid, makeStyles, Paper,
} from '@material-ui/core';
import useFetchBrands from '@/hooks/useFetchBrands';
import Header from '../Header';
import Loading from '../LoadingBackdrop';
import SelectYear from '../SelectYear';
import SelectBrand from '../SelectBrand';
import SelectModel from '../SelectModel';
import ButtonSearch from '../ButtonSearch';

const useStyles = makeStyles(({ spacing }) => ({
  containerForm: {
    padding: `${spacing(5)}px ${spacing(8)}px`,
    backgroundColor: '#FFF',
    borderRadius: '5px',
    width: '100%',
  },
}));

function FormAuto() {
  const classes = useStyles();
  const fetchBrands = useFetchBrands();

  useEffect(() => {
    fetchBrands();
  // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);

  return (
    <Box margin="auto 0" height="100%" width="100%">
      <Header />
      <Loading>
        <Paper className={classes.containerForm} elevation={2}>
          <Grid container spacing={3}>
            <SelectBrand />
            <SelectModel />
            <SelectYear />
            <ButtonSearch />
          </Grid>
        </Paper>
      </Loading>
    </Box>
  );
}

export default FormAuto;
