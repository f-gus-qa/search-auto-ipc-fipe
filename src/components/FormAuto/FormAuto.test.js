import React from 'react';
import { render } from '@testing-library/react';
import { create } from 'react-test-renderer';
import FormAuto from './FormAuto';
import useFetchBrands from '@/hooks/useFetchBrands';

jest.mock('@/hooks/useFetchBrands', () => jest.fn());
jest.mock('../Header', () => jest.fn().mockImplementation(
  () => <div data-testid="header" />,
));
jest.mock('../LoadingBackdrop', () => jest.fn().mockImplementation(
  ({ children }) => <div data-testid="loading">{children}</div>,
));
jest.mock('../SelectYear', () => jest.fn().mockImplementation(
  () => <div data-testid="select-year" />,
));
jest.mock('../SelectBrand', () => jest.fn().mockImplementation(
  () => <div data-testid="select-brand" />,
));
jest.mock('../SelectModel', () => jest.fn().mockImplementation(
  () => <div data-testid="select-model" />,
));
jest.mock('../ButtonSearch', () => jest.fn().mockImplementation(
  () => <div data-testid="button-search" />,
));

const fetchBrands = jest.fn();
useFetchBrands.mockReturnValue(fetchBrands);

describe('<FormAuto>', () => {
  it('checks renders structure', () => {
    const { toJSON } = create(<FormAuto />);

    expect(toJSON()).toMatchSnapshot();
  });

  it('checks useEffect call fetchBrands', () => {
    render(<FormAuto />);

    expect(fetchBrands).toHaveBeenCalledTimes(1);
  });
});
