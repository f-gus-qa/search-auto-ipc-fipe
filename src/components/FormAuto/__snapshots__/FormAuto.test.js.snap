// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormAuto> checks renders structure 1`] = `
<div
  className="MuiBox-root MuiBox-root-2"
>
  <div
    data-testid="header"
  />
  <div
    data-testid="loading"
  >
    <div
      className="MuiPaper-root makeStyles-containerForm-1 MuiPaper-elevation2 MuiPaper-rounded"
    >
      <div
        className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3"
      >
        <div
          data-testid="select-brand"
        />
        <div
          data-testid="select-model"
        />
        <div
          data-testid="select-year"
        />
        <div
          data-testid="button-search"
        />
      </div>
    </div>
  </div>
</div>
`;
